### PR TITLE
Update passwordValidator and passwordRequiredValidatorText prop

### PR DESCRIPTION
**Pull request by ektaghag-eaton**
Fixes BLUI 4703

#### Changes proposed in this Pull Request:

- Make `passwordValidator`  props accessible from the `LoginScreen` component.
- Remove `passwordRequiredValidatorText` from the `LoginScreen` component.
-

#### Screenshots / Screen Recording (if applicable)

-

#### To Test:

- Add below code snippet below in 
`passwordValidator={(password: string): boolean | string => {
            if (password?.length > 5) {
                return true;
            }
            return 'Password must be at least 6 characters';
        }}`

- go to http://localhost:3000/login
- enter a password less than 6 characters login button will be disabled
- blur password input field will show an error message `Password must be at least 6 characters`

#### Any specific feedback you are looking for?

-

### CODE REVIEW

File: login-workflow/src/screens/LoginScreen/LoginScreen.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ const EMAIL_REGEX = /^[A-Z0-9._%+'-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
  * @category Component
  */
 
-type LoginScreenPropsPublic = Omit<LoginScreenProps, 'passwordValidator'> & { passwordRequiredValidatorText?: string };
```
Comment (surajeaton) [inline login-workflow/src/screens/LoginScreen/LoginScreen.tsx:46]: If we are not using `passwordValidator` prop, it is better to Omit. Please revert this changes

File: login-workflow/src/screens/LoginScreen/LoginScreen.tsx
Comment (surajeaton) [inline login-workflow/src/screens/LoginScreen/LoginScreen.tsx:1]: passwordValidator should be removed from js doc.

Comment (surajeaton) [inline login-workflow/src/screens/LoginScreen/LoginScreen.tsx:1]: We should use passwordValidator prop to override the default value of passwordValidator

**Comment by ektaghag-eaton** [comment]
@surajeaton , please review.

**Review by daileytj — APPROVED** [review]
